### ISSUE TITLE
azurerm v2.61.0.5_x5 plugin crash re-deploying resources

### ISSUE THREAD

**Issue by johncclayton** [hashicorp/terraform-provider-azurerm#12009]
When deleting a CDN endpoint resource in Azure... 

Error: The terraform-provider-azurerm_v2.61.0_x5.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

PS C:\Users\john\src\appcast-storage\terraform> terraform -v

Terraform v0.15.4
on windows_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.61.0

### Affected Resource(s)

* azurerm_cdn_endpoint

### Terraform Configuration Files

```hcl
resource "azurerm_cdn_endpoint" "endpoint" {
  name                = "${var.storage_account_name}"
  profile_name        = azurerm_cdn_profile.cdn.name
  resource_group_name = azurerm_resource_group.rg.name
  location            = "Global"
  is_http_allowed     = false
  is_https_allowed    = true
  origin_path         = "/${var.container_name}"

  origin {
    name              = var.storage_account_name
    host_name         = "${var.storage_account_name}.blob.core.windows.net"
  }
}
```

### Debug Output

https://gist.github.com/johncclayton/521473a577a5b80876a4231d48fc5783

### Expected Behaviour

Resources deleted and re-created (the origin data was changed within the endpoint).

### Actual Behaviour

Terraform command crashed.

### Steps to Reproduce

Running the "apply" command reproduces the error.

1. `terraform apply`

### Important Factoids

I have used TF to create the resource group, storage account already.  Then I created a CDN endpoint and configured the origin and custom DNS data.

I'm now trying to bring back my manual changes under the control of terraform.  The manual changes are just the CDN endpoint, it's origin and custom DNS settings. 

### References

* #0000

**Comment by chris-hibberd** [comment]
I encountered the same issue when running terraform destroy against an Azure CDN endpoint that had been created via Terraform and manually provisioned with an HTTPS custom domain (via the portal). I then tried to manually delete the CDN endpoint via the Azure Portal. This produced an error until the corresponding DNS records were removed from our DNS provider. Since the DNS records are not managed via Terraform, I suspect that it will  might not be possible to delete the CDN endpoint

**Comment by meip** [comment]
Azure is blocking the deletion of a CDN profile/endpoint if there are CName records pointing to the endpoint. If you try to do this by hand in the portal you get this error

> Cannot delete custom domain "blabla.xy.com" because it is still directly or indirectly (using "cdnverify" prefix) CNAMEd to CDN endpoint "blabla.azureedge.net". Please remove the DNS CNAME record and try again.

So as described by @chris-hibberd manually removing the CName record solved the issue for me and deleting the CDN with Terraform just worked.

**Comment by lonegunmanb** [comment]
I believe it has been resolved by 90cc78576003f09769b3260f3e41caf5910a4c00 .

**Comment by rcskosir** [comment]
Thanks for opening this issue. This was a problem in the 2.x version of the provider which is no longer actively maintained.
If this is still an issue with the 3.x version of the provider please do let us know by opening a new issue, thanks!
